samples/third-party-lib/src/components/ThreeDimensionalGraph/ThreeDimensionalGraphModel.ts

```diff
@@ -0,0 +1,226 @@
+import { Feature } from "@vertigis/arcgis-extensions/data/Feature";
+import { FeatureStream } from "@vertigis/arcgis-extensions/data/FeatureStream";
+import { TableExtension } from "@vertigis/arcgis-extensions/data/TableExtension";
+import { QueryService } from "@vertigis/arcgis-extensions/tasks/query/QueryService";
+import {
+    ComponentModelBase,
+    serializable,
+    PropertyDefs,
+    ComponentModelProperties,
+} from "@vertigis/web/models";
+import { BrandingService } from "@vertigis/web/branding/BrandingService";
+import { command, Features } from "@vertigis/web/messaging";
+import { inject, FrameworkServiceType } from "@vertigis/web/services";
+import type {
+    GraphData as ForceGraphData,
+    LinkObject,
+    NodeObject as ForceNodeObject,
+} from "force-graph";
+import { toFeatureArray } from "./utils";
+
+interface ThreeDimensionalGraphModelProperties
+    extends ComponentModelProperties {
+    surveyTableExtension?: TableExtension;
+}
+
+export interface SurveyorNodeObject extends ForceNodeObject {
+    id: string;
+    type: "surveyor";
+}
+
+export interface SurveyNodeObject extends ForceNodeObject {
+    id: number;
+    type: "survey";
+    date: number;
+    hydrantNum: string;
+    result: "PASS" | "FAIL";
+}
+
+export type NodeObject = SurveyorNodeObject | SurveyNodeObject;
+
+export interface GraphData extends ForceGraphData {
+    nodes: NodeObject[];
+    links: LinkObject[];
+}
+
+@serializable
+export default class ThreeDimensionalGraphModel extends ComponentModelBase<
+    ThreeDimensionalGraphModelProperties
+> {
+    @inject(FrameworkServiceType.BRANDING)
+    brandingService: BrandingService | undefined;
+
+    // This will be populated from the app config item reference.
+    surveyTableExtension: TableExtension | undefined;
+
+    selectedSurveyId: number | undefined;
+    hydrants: Record<string, Feature> = {};
+    surveys: Record<number, Feature> = {};
+    graphData: GraphData = { links: [], nodes: [] };
+
+    getNodeLabel = (node: NodeObject): string => {
+        if (node.type === "surveyor") {
+            return `Surveyor: ${node.id}`;
+        }
+
+        return `Hydrant Survey<br />
+        Survey ID: ${node.id}<br />
+        Hydrant number: ${node.hydrantNum}<br />
+        Result: ${node.result}<br />
+        Date: ${new Date(node.date).toLocaleString()}`;
+    };
+
+    getNodeColor = (node: NodeObject): string => {
+        if (node.type === "surveyor") {
```

Comment:
Minor style note: when branches are mutually exclusive (exactly one will be chosen), I think that `if ... else if ... else` or `switch` is a better way to structure the code rather than independent `if` statements. It just alerts the reader immediately that this is will happen, without having to analyze the code. Does that make sense?

Reply:
Updated.